server/sandbox_run_linux.go

```diff
@@ -1091,7 +1091,43 @@ func (s *Server) configureGeneratorForSysctls(ctx context.Context, g *generate.G
 		g.AddLinuxSysctl(key, value)
 		sysctlsToReturn[key] = value
 	}
-	return sysctlsToReturn
+	return configurePingGroupRangeGivenIDMappings(ctx, g, sandboxIDMappings, sysctlsToReturn)
+}
+
+func configurePingGroupRangeGivenIDMappings(ctx context.Context, g *generate.Generator, sandboxIDMappings *idtools.IDMappings, sysctls map[string]string) map[string]string {
+	// We have to manually fuss with this specific sysctl.
+	// It's commonly set to the max range by default "0 2147483647".
+	// However, a pod with GIDMappings may not actually have the upper range set,
+	// which means attempting to set this sysctl will fail with EINVAL
+	// Instead, update the max of the group range to be the largest group value in the IDMappings.
+	const (
+		pingGroupRangeKey        = "net.ipv4.ping_group_range"
+		pingGroupFullRangeBottom = "0"
+		pingGroupFullRangeTop    = "2147483647"
+	)
+	val, ok := sysctls[pingGroupRangeKey]
+	if !ok || sandboxIDMappings == nil {
+		return sysctls
+	}
+	// Only do this if the value is `0 2147483647`
+	currentRange := strings.Fields(val)
+	if len(currentRange) != 2 || currentRange[0] != pingGroupFullRangeBottom || currentRange[1] != pingGroupFullRangeTop {
+		return sysctls
+	}
+
+	maxID := 0
+	for _, mapping := range sandboxIDMappings.GIDs() {
+		topOfRange := mapping.ContainerID + mapping.Size - 1
+		if maxID < topOfRange {
+			maxID = topOfRange
+		}
+	}
```

Comment:
The recent Go version has the `max()` built-in:

```go
	var maxID int
	for _, mapping := range sandboxIDMappings.GIDs() {
		maxID = max(maxID, mapping.ContainerID + mapping.Size - 1)
	}
```

Reply:
added a follow up https://github.com/cri-o/cri-o/pull/8184